addon/modules/pagination/_default_future.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable */
+
+export default {
+  prepareQueryObject: function () {
+    return {
+      filterQueryParam: this.get('filterQueryParam'),
+      sortQueryParam: this.get('sortQueryParam'),
+      pageQueryParam: {
+        nextOffset: this.get('__lastOffset'),
+        pageSize: this.get('__pageSize')
+      }
+    }
+  },
+
+  requestNext: function (queryObject, serializer) {
+    let dataKey = this.get('objectBrowserConfig.listConfig.items')
+    serializer.query(queryObject).then((response) => {
+      this.set(dataKey, response)
+    })
+  },
+
+  setPageQueryParam: function (controller) {
+    let nextOffset = controller.get('__lastOffset')
+    let size = controller.get('__size')
+    let pageObject = {
+      nextOffset,
+      size
+    }
+
+    let normalizedPageObject = this.normalizePage(pageObject)
+    controller.set('pageQueryParam', normalizedPageObject)
+  },
+
+  setPageQueryParamForLoadPrevious: function (controller) {
+    debugger;
+    let nextOffset = controller.get('__firstOffset')
+    let size = controller.get('__size')
+    let pageObject = {
+      nextOffset: nextOffset - size,
+      size
+    }
+
+    let normalizedPageObject = this.normalizePage(pageObject)
+
+    //let dataKey = controller.get('objectBrowserConfig.listConfig.items')
```

Comment:
maybe add a comment to the top of the file to indicate the purpose/plan in case someone else has to take over?